examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -115,14 +118,62 @@ function Initialise_DYCOMS!(state::Vars, aux::Vars, (x,y,z), t)
   state.moisture.ρq_tot = ρ * q_tot
 end   
 
+# TODO: temporary; move to new CLIMA module
+function gather_diags(dg, Q)
+  bl = dg.balancelaw
+  grid = dg.grid
+  topology = grid.topology
 
-function run(mpicomm, ArrayType, dim, topl, N, timeend, DT, dt, C_smag, LHF, SHF, C_drag, zmax, zsponge)
+  dim = dimensionality(grid)
+  N = polynomialorder(grid)
+  Nq = N + 1
+  Nqk = dim == 2 ? 1 : Nq
+  Nfp = Nq * Nqk
+  nrealelem = length(topology.realelems)
+
+  DFloat = eltype(Q)
+
+  nstate = 1 # temporarily, \rho only
+  # nstate = num_state(bl, DFLoat)
 
+  mpirank = MPI.Comm_rank(MPI.COMM_WORLD)
+  nranks = MPI.Comm_size(MPI.COMM_WORLD)
+
+  host_array = Array ∈ typeof(Q).parameters
+  localQ = host_array ? Q.realQ : Array(Q.realQ)
```

Comment:
```suggestion
  localQ = host_array ? Q.realdata : Array(Q.realdata)
```
will be required once #470 is merged.